collectd/collectd.go

```diff
@@ -0,0 +1,64 @@
+package collectd
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"math"
+	"time"
+
+	"github.com/kimor79/gollectd"
+)
+
+var ErrTimeOutOfBounds = errors.New("The time is to large to store")
+
+//TODO corylanou: This needs to be made a public `main.Point` so we can share this across packages.
+type Metrics []Metric
+type Metric struct {
+	Name      string
+	Tags      map[string]string
+	Value     interface{}
+	Timestamp time.Time
+}
+
+func Unmarshal(data *gollectd.Packet) (Metrics, error) {
```

Comment:
If you mean *Metrics: in Go, there's almost never justification for returning a pointer directly to: a slice, a map, a channel, or a func.

In this case, []Metric is a lot better than []*Metric too, unless you have a very specific reason to do otherwise.